### PR TITLE
Object Catalog

### PR DESCRIPTION
**Pull request by hmert**
Related #1559

### CODE REVIEW

File: src/Ojs/CoreBundle/Filter/CatalogFilter.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            $mappings['catalogs']['targetEntity'] !== 'Ojs\JournalBundle\Entity\Catalog'){
+            return '';
+        }
+
+        //return if journal filter disabled globally for current entity
+        if(isset($GLOBALS[$targetEntity->getName().'#catalogFilter']) && $GLOBALS[$targetEntity->getName().'#catalogFilter'] == false){
+            return '';
+        }
+
+        $addCondSql = $targetTableAlias.'.array_catalogs LIKE \'%'.$this->catalogs[0].'%\'';
```
Comment (o) [inline src/Ojs/CoreBundle/Filter/CatalogFilter.php]: Looks like, this only manipulates table name which is generated by doctrine query builder.

File: src/Ojs/CoreBundle/Filter/CatalogFilter.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    public function addFilterConstraint(ClassMetadata $targetEntity, $targetTableAlias)
+    {
+        //if params have asterisk sign, pass catalog filter
+        if(in_array('*', $this->catalogs)){
+            return '';
+        }
+
+        $mappings = $targetEntity->getAssociationMappings();
+        if(!array_key_exists('catalogs', $mappings) ||
```
Comment (o) [inline src/Ojs/CoreBundle/Filter/CatalogFilter.php]: For my opinion, checking eligibility of entity via explicit inheritance is a better choice.

Like `$targetEntity->reflClass->implementsInterface('CatalogAware')`

File: src/Ojs/CoreBundle/Filter/CatalogFilter.php (hunk 1 above)
Comment (hmert) [inline src/Ojs/CoreBundle/Filter/CatalogFilter.php]:
```SQL
SELECT bar, name
FROM foobar
INNER JOIN names ON foobar.bar = names.value
WHERE foo = 1 and bar = ANY (1,2,3)
```

We need to change LIKE query to ANY.

**Comment by behram** [comment]
development about this branch probably continues. so i will close this pr for now

**Comment by behram** [comment]
development will resumes from https://github.com/ojs/ojs/tree/feature/object-catalog-2 branch

**Pull request closed without merging**